### PR TITLE
Fix parent scope filter

### PR DESCRIPTION
**Pull request by Stef-Rousset**
#### :tophat: Description
In an assembly on proposals index view, sometimes  in the filters, a scope is not checked because its children are not checked while they should be all checked when loading the page.

#### :pushpin: Related Issues
*Link your PR to an issue*
- [Notion card](https://www.notion.so/opensourcepolitics/bd4bf860eab94f9daca7da803f09e3b2?v=c96df65755dd403d95e86f82d7749709&p=1988f63b9b3e4cb6abd1e2b4bcbc53cc&pm=c)

#### Testing
*Describe the best way to test or validate your PR.*

1. As an admin, create an assembly with scopes enabled, choose a scope that has subscopes (like "Indiana"), or choose one assembly that has scopes enabled and a scope.
2. As an admin, go to parameters > scopes > Indiana, click on one of its subscopes (like [Grimesstad) and  add subscopes to it
3. As a user, go to the assemblies > assembly > proposals and see in the filters that all the scopes and their children are checked

#### Tasks
- [X ] Add specs

### :camera: Screenshots
*Please add screenshots of the changes you're proposing if related to the UI*

### CODE REVIEW

**Review by BarbaraOliveira13 — DISMISSED** [review]
Ran it, works 👍 👏

File: lib/extends/controllers/decidim/proposals/proposals_controller_extends.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+module ProposalsControllerExtends
+  extend ActiveSupport::Concern
+  included do
+    private
+
+    def default_filter_scope_params
```
Comment (Quentinchampenois) [inline lib/extends/controllers/decidim/proposals/proposals_controller_extends.rb:9]:
```suggestion
        def default_filter_scope_params
      return "all" unless current_component.scopes.any?

      scope = current_component.scope
      return current_component_scope_ids if scope.blank?

      subscope_ids = scope.children&.map(&:id)
      ids = ["all", scope.id] + subscope_ids
      ids += Decidim::Scope.where(parent_id: subscope_ids).pluck(:id) if subscope_ids.present?
      ids.map(&:to_s)
    end

    def current_component_scope_ids
      component_scopes = current_component.scopes.pluck(:id).map(&:to_s)

      %w(all global) + component_scopes
    end
```

Comment (Quentinchampenois) [reply lib/extends/controllers/decidim/proposals/proposals_controller_extends.rb:9]: This a refactor suggestion, it works the same way you did

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Hey ! It works well, I tested with 4 levels of subscopes and it is already checked as expected ! 

I just have a refactor suggestion to reduce the level of indentation and perceived complexity

**Pull request merged**